apps/api/v2/src/ee/calendars/input/create-ics.input.ts

```diff
@@ -0,0 +1,52 @@
+import { ApiProperty } from "@nestjs/swagger";
+import {
+  ArrayNotEmpty,
+  IsBoolean,
+  IsOptional,
+  Validate,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+} from "class-validator";
+import { IsNotEmpty, IsArray } from "class-validator";
+
+// Custom constraint to validate ICS URLs
+@ValidatorConstraint({ async: false })
+export class IsICSUrlConstraint implements ValidatorConstraintInterface {
+  validate(url: unknown) {
+    if (typeof url !== "string") return false;
+
+    // Check if it's a valid URL and ends with .ics
+    try {
+      const urlObject = new URL(url);
+      return (
+        urlObject.protocol === "http:" ||
+        (urlObject.protocol === "https:" && urlObject.pathname.endsWith(".ics"))
+      );
+    } catch (error) {
+      return false;
+    }
+  }
+
+  defaultMessage() {
+    return "The URL must be a valid ICS URL (ending with .ics)";
+  }
+}
+
+export class CreateIcsFeedInputDto {
+  @IsArray()
+  @ArrayNotEmpty()
+  @IsNotEmpty({ each: true })
+  @Validate(IsICSUrlConstraint, { each: true }) // Apply the custom validator to each element in the array
```

Comment:
Need to add `@ApiProperty({` documentation here too - would add example URL and explain that it has to start with http / https and end with ics.